dhis-2/dhis-api/src/main/java/org/hisp/dhis/programstagefilter/EventDataFilter.java

```diff
@@ -170,8 +170,10 @@ public String getNull() {
     return nullFilter;
   }
 
-  public void setNull(String isNull) {
-    this.nullFilter = isNull == null ? "" : isNull;
+  public void setNull(String nullFilter) {
```

Comment:
I thought this is going to be a boolean? 🤔

Reply:
yes but didn't we say we only want to store it when it's true?
if it was a boolean, it'd be false by default and stored in the database
if it's a String, we can just set it to null by default, so it'll be excluded by Jackson, unless we explicitly set it to "true"

Reply:
> yes but didn't we say we only want to store it when it's true?
I think so but why does that mean we accept/require `{"null": ""}`? I though we are going for the not perfect but slightly better `{"null": true}` if users want to filter for null.

@enricocolasante maybe I misremember 😅

Reply:
Didn't we agree that we accept any value there? `{"null": [ANYTHING_BUT_NULL]}`?
Then when saving we remove the value there and use `true`

Reply:
Why can't we use Boolean (which is nullable). We use it for "followUp" as well.

Reply:
Atleast the values would then be limited to only three valid values, true/false/null.